### PR TITLE
Extend default skip magic rather than overwrite it.

### PR DESCRIPTION
**Pull request by qkaiser**
unblob has a decent default skip magic list that gets overwritten if a user provides its own, which means unblob users need to redefine all of unblob's default skip magic through the CLI whenever they provide their own.

Changed the logic so that user provided skip magic values are simply appended to unblob's default list.

This is an opinonated change, but we think unblob default skip magic is well thought out and the inability to fully overwrite it should not be a problem for the vast majority of our user base.

Related to #647

### CODE REVIEW

**Review by e3krisztian — CHANGES_REQUESTED** [review]
The intention behind replacing the skip list is that by doing so we can force extraction one of the file types on this list, even if that would be skipped by default.

Changing skip magic specification to extend only would prevent clearing this magic-skip list, so unblob would be unable to extract file types that are on its magic-skip list, even if technically it has a handler for it.

Maybe a better extension would be to add a new `--extend-default-skip-magic` bool option, which would add the `DEFAULT_SKIP_MAGIC` to the user supplied list.

This solution would let the users extend the magic-skip list, while not changing the current behavior.

-----

Alternatively we could have a `--clear-skip-magic` bool option, which would be needed to get the current behavior (on `main`), replacing the magic skip list. Without `--clear-skip-magic` the `DEFAULT_SKIP_MAGIC` would be added to the user supplied (`--skip-magic`) magics.

I would prefer this alternative, as I think adding a new magic is more common/natural than replacing the whole list, even if it is an `unblob` behavior change.

I think this `--clear-skip-magic` alternative would be easier to understand and use, than the `--extend-default-skip-magic` variant.

File: unblob/processing.py [hunk 1]
```diff
@@ -86,7 +87,10 @@ class ExtractionConfig:
     entropy_depth: int
     entropy_plot: bool = False
     max_depth: int = DEFAULT_DEPTH
-    skip_magic: Iterable[str] = DEFAULT_SKIP_MAGIC
+    skip_magic: Iterable[str] = attr.field(
+        default=DEFAULT_SKIP_MAGIC,
+        converter=lambda value: list(itertools.chain(DEFAULT_SKIP_MAGIC, value)),
+    )
```
[the remark this answers is not in the document]
Comment (e3krisztian) [reply unblob/processing.py:93]: Maybe it is also plural `--clear-skip-magics`?

File: unblob/processing.py [hunk 2]
```diff
@@ -49,7 +50,7 @@
 
 DEFAULT_DEPTH = 10
 DEFAULT_PROCESS_NUM = multiprocessing.cpu_count()
-DEFAULT_SKIP_MAGIC = (
+DEFAULT_SKIP_MAGIC: Iterable[str] = [
```
[the remark this answers is not in the document]
Comment (qkaiser) [reply unblob/processing.py]: Reverted the change.

File: unblob/processing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (qkaiser) [reply unblob/processing.py]: Reverted. Moved setting `skip_magic` to `cli.py`. Using plural form.

File: unblob/cli.py [hunk 3]
```diff
@@ -263,6 +272,11 @@ def cli(
     extra_dir_handlers = plugin_manager.load_dir_handlers_from_plugins()
     dir_handlers += tuple(extra_dir_handlers)
 
+    if clear_skip_magics and skip_magic is not DEFAULT_SKIP_MAGIC:
+        skip_magic = tuple(sorted(set(skip_magic).difference(DEFAULT_SKIP_MAGIC)))
+    else:
+        skip_magic = tuple(sorted(set(skip_magic).union(DEFAULT_SKIP_MAGIC)))
+
```
Comment (e3krisztian) [inline unblob/cli.py:279]: `difference` is a bit stronger than needed: we just do not want to add the defaults, but when added by a user, we do not want to remove them.
```suggestion
    extra_magics_to_skip = () if clear_skip_magics else DEFAULT_SKIP_MAGIC
    skip_magic = tuple(sorted(set(skip_magic).union(extra_magics_to_skip)))
```

**Review by e3krisztian — CHANGES_REQUESTED** [review]

File: unblob/cli.py (hunk 3 above)
Comment (qkaiser) [reply unblob/cli.py:279]: Thanks. I actually struggled with this one, my brain was a bit foggy ^^

Comment (qkaiser) [reply unblob/cli.py]: Tests are failing if I implement your change. This is due to the fact that regardless of `clear_skip_magics`, `skip_magic` defaults to `DEFAULT_SKIP_MAGIC` if the user does not explicitly provide others.

From my perspective, using `--clear-skip-magics` without any `--skip-magic` means the default skip_magic should be cleared. See the test at https://github.com/onekey-sec/unblob/pull/693/files#diff-4e8715c7a425ee52e74b7df4d34efd32e8c92f3e60bd51bc2e1ad5943b82032eR386.

**Review by e3krisztian — APPROVED** [review]
This is a backward incompatible behavior change for the `--skip-magic` option.

The original way was somewhat surprising and harder to use (#647).

**Pull request merged**